cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -1575,6 +1575,9 @@ protected List<Container> getInitContainers(ImagePullPolicy imagePullPolicy) {
 
     @Override
     protected List<Container> getContainers(ImagePullPolicy imagePullPolicy) {
+        String livenessCommand = useKRaft ? "/opt/kafka/kafka_liveness_kraft.sh" : "/opt/kafka/kafka_liveness.sh";
+        // In ZooKeeper mode kafka-agent will create /var/opt/kafka/kafka-ready in the container
+        String[] readinessCommand = useKRaft ? new String[]{"/opt/kafka/kafka_readiness_kraft.sh"} : new String[]{"test", "-f", "/var/opt/kafka/kafka-ready"};
```

Comment:
I hadn't, I hadn't spotted the `STRIMZI_KRAFT_ENABLED` env var, so if we can use that it would mean keeping all the ZooKeeper vs KRaft logic in the scripts which I agree would be cleaner

Reply:
I've updated this to use a single script for liveness and a single script for readiness.